### PR TITLE
[New Model] GlobalTransportLabel 1.0.0

### LINKED ISSUE #431: [New Model]: transport unit / global transport label
## Model Description
We want to attach the information of a global transport label to a digital twin that represents a transport unit. 

## Relevant Standards
VDA 4994 Global Transport Label https://www.vda.de/de/aktuelles/publikationen/publication/vda-4992---mat-label-und-emat-datenaustausch

## Example Data
{
  "labelType": "M",
  "shipFrom": {
    "BPNA": "yedUsFwdkelQbxeTeQOvaScfqIOOmaa",
    "addess": "eOMtThyhVNLWUZNRcBaQKxI",
    "supplierId": "RYtGKbgicZaHCBRQDSx",
    "vatNumber": "JxkyvRnL"
  },
  "customerReference1": {
    "deliveryNote": "QvBQYuxiXXVytGCxzVllpgTJKhRQq",
    "supplierNumber": "qszYL"
  },
  "shipTo": {
    "BPNA": "UfzQhdgLLfDTDGspDb",
    "addess": "gNfZBdyFGRajVfJNonEnOinZj",
    "unloadingPoint": "UMaAIKKIkknjWEXJUfPxxQHeWKEJ",
    "internalDestination": "dpHYZGhtgdntugzvvKAXLhM",
    "plant": "VLhpfQGTMDYpsBZxvfBoeygjb",
    "vatNumber": "LlN"
  }
...
}

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by jonbckr**
## Description
We want to attach the information of a global transport label to a digital twin that represents a transport unit. This PR represents the Aspect Model "GlobalTransportLabel" and consists of most data properties to fulfill the following label: 

image

 -->

Closes #431 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.1)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by jonbckr** [comment]
@jacewski-bosch Before we start MS2, please revise the model with an additional focus on the following details: 
- SpecificAssetIDs for address (address_characteristic), businessPartner (business_partner_number), quantity (quantity), catenaXId (uuid)
- The lists including the character limitations for the properties "optionalInformation" and "customerReference3"
- The RegEx for "date", and "licencePlate"

**Comment by jonbckr** [comment]
modeling_team

File: io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+   samm:preferredName "Supplier Number"@en ;
+   samm:description "Supplier number assigned to the seller by the customer. This supplier number may differ from ship-from's identifier, depending on the numbering concept applied by the customer."@en ;
+   samm:characteristic :SupplierNumberCharacteristic ;
+   samm:exampleValue "987654321" .
+
+:expectedTimeOfArrival a samm:Property ;
+   samm:preferredName "ExpectedTime Of Arrival"@en ;
+   samm:description "Expected/request delivery time of the goods at the customer's premises."@en ;
+   samm:characteristic :ExpectedTimeOfArrivalCharacteristic ;
+   samm:exampleValue "2023-10-27"^^xsd:dateTime .
```
Comment (agg3fe) [inline io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl:250]: example value be in date and time?

File: io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+   samm:dataType xsd:string .
+
+:LicencePlateRegEx a samm-c:RegularExpressionConstraint ;
+   samm:preferredName "Licence Plate RegEx"@en ;
+   samm:description "Regular Expression to ensure the correct format of the Licence Plate"@en ;
+   samm:value "^[a-zA-Z0-9]{2}\\s[a-zA-Z0-9]{4,9}\\s\\d{9}$" .
+
+:DateCategoryEntity a samm:Entity ;
+   samm:preferredName "Date Category Entity"@en ;
+   samm:description "Encapsualtes the information of the Date Category Entity."@en ;
```
Comment (agg3fe) [inline io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl:446]: typo `Encapsualtes`

File: io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+   samm:description "Encapsulation of Customer References."@en ;
+   samm:properties ( :deliveryNoteNumber :supplierNumber ) .
+
+:LogisticsReferenceEntity a samm:Entity ;
+   samm:preferredName "Logistics Reference Entity"@en ;
+   samm:description "Logistics information about estimated time of arrival, quantity and weight."@en ;
+   samm:properties ( :expectedTimeOfArrival :netWeight :grossWeight ext-quantity:itemQuantityProperty ) .
+
+:ContentDescriptionEntity a samm:Entity ;
+   samm:preferredName "Content Description Entity"@en ;
```
Comment (agg3fe) [inline io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl:182]: `description` missing

File: io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#> .
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.global_transport_label:1.0.0#> .
+@prefix ext-address_characteristic: <urn:samm:io.catenax.shared.address_characteristic:3.0.0#> .
+@prefix ext-business_partner_number: <urn:samm:io.catenax.shared.business_partner_number:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl:35]: want to include business_partner_number:2.0.0?

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

File: io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl (hunk 1 above)
Comment (jonbckr) [reply io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl]: Done: changed to only date.

File: io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+   samm:dataType xsd:double .
+
+:GrossWeight a samm:Characteristic ;
+   samm:preferredName "Gross Weight"@en ;
+   samm:description "Gross weight of the loading unit or inner packaging item in kilogram."@en ;
+   samm:dataType xsd:double .
+
+:CustomerPartNumberCharacteristic a samm:Characteristic ;
+   samm:preferredName "Customer part number characteristic"@en ;
+   samm:description "Decribes the number of the part provided by the customer."@en ;
```
[the remark this answers is not in the document]
Comment (jonbckr) [reply io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl]: done.

File: io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl (hunk 3 above)
Comment (jonbckr) [reply io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl:182]: done.

File: io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl (hunk 4 above)
Comment (jonbckr) [reply io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl]: Yes.

**Comment by jonbckr** [comment]
@agg3fe All comments are checked. Will start with MS3.

File: io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl [hunk 6]
```diff
@@ -0,0 +1,483 @@
...
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#> .
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.global_transport_label:1.0.0#> .
+@prefix ext-address_characteristic: <urn:samm:io.catenax.shared.address_characteristic:3.0.0#> .
+@prefix ext-business_partner_number: <urn:samm:io.catenax.shared.business_partner_number:2.0.0#> .
+@prefix ext-quantity: <urn:samm:io.catenax.shared.quantity:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl:36]: FYI, Quantity 2.0.0 and Uuid 2.0.0 are also released. if you want you can include them.

**Review by agg3fe — APPROVED** [review]
LGTM

File: io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl (hunk 6 above)
Comment (jonbckr) [reply io.catenax.global_transport_label/1.0.0/GlobalTransportLabel.ttl:36]: will do that!

**Pull request merged**